pkg/util/fedinformer/typedmanager/multi-cluster-manager_test.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package typedmanager
+
+import (
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/karmada-io/karmada/pkg/util/fedinformer"
+)
+
+func TestMultiClusterInformerManager(t *testing.T) {
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	transforms := map[schema.GroupVersionResource]cache.TransformFunc{
+		nodeGVR: fedinformer.NodeTransformFunc,
+		podGVR:  fedinformer.PodTransformFunc,
+	}
+
+	manager := NewMultiClusterInformerManager(stopCh, transforms)
+
+	t.Run("ForCluster", func(_ *testing.T) {
+		cluster := "test-cluster"
+		client := fake.NewSimpleClientset()
+		resync := 10 * time.Second
+
+		singleManager := manager.ForCluster(cluster, client, resync)
+		if singleManager == nil {
+			t.Fatalf("ForCluster() returned nil")
+		}
+
+		if !manager.IsManagerExist(cluster) {
+			t.Fatalf("IsManagerExist() returned false for existing cluster")
+		}
+	})
+
+	t.Run("GetSingleClusterManager", func(t *testing.T) {
+		cluster := "test-cluster"
+		singleManager := manager.GetSingleClusterManager(cluster)
+		if singleManager == nil {
+			t.Fatalf("GetSingleClusterManager() returned nil for existing cluster")
+		}
+
+		nonExistentCluster := "non-existent-cluster"
+		singleManager = manager.GetSingleClusterManager(nonExistentCluster)
+		if singleManager != nil {
+			t.Fatalf("GetSingleClusterManager() returned non-nil for non-existent cluster")
+		}
+	})
+
+	t.Run("Start and Stop", func(t *testing.T) {
+		cluster := "test-cluster-2"
+		client := fake.NewSimpleClientset()
+		resync := 10 * time.Second
+
+		manager.ForCluster(cluster, client, resync)
+		manager.Start(cluster)
+
+		manager.Stop(cluster)
+
+		if manager.IsManagerExist(cluster) {
+			t.Fatalf("IsManagerExist() returned true after Stop()")
+		}
+	})
```

Comment:
Aren't methods `WaitForCacheSync` and `WaitForCacheSyncWithTimeout` easy to test?

Reply:
Yes, I'll add them up.